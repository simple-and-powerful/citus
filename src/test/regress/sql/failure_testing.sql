CREATE TABLE test (a int, b int);
SELECT create_distributed_table('test', 'a');

BEGIN;
UPDATE test SET b = 2 WHERE a = 2;
ROLLBACK;

SELECT citus.mitmproxy('flow.contains(b"UPDATE").kill()');

BEGIN;
UPDATE test SET b = 2 WHERE a = 2;
ROLLBACK;

SELECT citus.mitmproxy('flow.contains(b"BEGIN").kill()');

BEGIN;
UPDATE test SET b = 2 WHERE a = 2;
ROLLBACK;

SET citus.multi_shard_commit_protocol = '2pc';
SELECT citus.mitmproxy('flow.contains(b"COMMIT").kill()');

BEGIN;
UPDATE test SET b = 2 WHERE a = 2;
COMMIT;

SELECT citus.mitmproxy('flow.allow()');

SET citus.multi_shard_commit_protocol = '2pc';

BEGIN;
UPDATE test SET b = 2 WHERE a = 2;
COMMIT;

DROP TABLE test;
