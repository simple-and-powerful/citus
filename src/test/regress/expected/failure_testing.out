CREATE TABLE test (a int, b int);
SELECT create_distributed_table('test', 'a');
 create_distributed_table 
--------------------------
 
(1 row)

BEGIN;
UPDATE test SET b = 2 WHERE a = 2;
ROLLBACK;
SELECT citus.mitmproxy('flow.contains(b"UPDATE").kill()');
 mitmproxy 
-----------
 
(1 row)

BEGIN;
UPDATE test SET b = 2 WHERE a = 2;
WARNING:  server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
CONTEXT:  while executing command on localhost:57640
ROLLBACK;
SELECT citus.mitmproxy('flow.contains(b"BEGIN").kill()');
 mitmproxy 
-----------
 
(1 row)

BEGIN;
UPDATE test SET b = 2 WHERE a = 2;
WARNING:  connection not open
CONTEXT:  while executing command on localhost:57640
ROLLBACK;
SET citus.multi_shard_commit_protocol = '2pc';
SELECT citus.mitmproxy('flow.contains(b"COMMIT").kill()');
 mitmproxy 
-----------
 
(1 row)

BEGIN;
UPDATE test SET b = 2 WHERE a = 2;
WARNING:  connection not open
CONTEXT:  while executing command on localhost:57640
COMMIT;
WARNING:  connection not open
CONTEXT:  while executing command on localhost:57640
WARNING:  connection not open
CONTEXT:  while executing command on localhost:57640
WARNING:  connection not open
CONTEXT:  while executing command on localhost:57640
SELECT citus.mitmproxy('flow.allow()');
 mitmproxy 
-----------
 
(1 row)

SET citus.multi_shard_commit_protocol = '2pc';
BEGIN;
UPDATE test SET b = 2 WHERE a = 2;
COMMIT;
DROP TABLE test;
